azurerm/internal/services/media/media_streaming_endpoint_resource.go

```diff
@@ -0,0 +1,591 @@
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"address": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"name": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"subnet_prefix_length": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"cdn_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+
+			"cdn_profile": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-A-Za-z0-9]{1,120}$"),
+					"CDN profile must be 1 - 120 characters long, can contain only letters, numbers, and hyphens.",
+				),
+			},
+
+			"cdn_provider": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"StandardVerizon",
+				}, false),
+			},
+
+			"cross_site_access_policies": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"client_access_policy": {
+							Type:         schema.TypeString,
+							Computed:     true,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"cross_domain_policy": {
+							Type:         schema.TypeString,
+							Computed:     true,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+
+			"custom_host_names": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"max_cache_age_seconds": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(1, 2147483647),
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceMediaStreamingEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Media.StreamingEndpointsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	streamingEndpointName := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	accountName := d.Get("media_services_account_name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	scaleUnits := d.Get("scale_units").(int)
+	autoStart := utils.Bool(false)
```

Comment:
we need to check for the presence of an existing resource (and use an ID Formatter) here, could make this:

```suggestion
	subscriptionId := meta.(*clients.Client).Account.SubscriptionID
	resourceId := parse.NewStreamingEndpointId(subscriptionId, d.Get("resource_group_name").(string), d.Get("media_services_account_name").(string), d.Get("name").(string))
	existing, err := client.Get(resourceId.ResourceGroup, resourceId.MediaserviceName, resourceId.Name)
	if err != nil {
		if !utils.ResponseWasNotFound(existing.Response()) {
			return fmt.Errorf("checking for presence of %s: %+v", resourceId, err)
		}
	}
	if !utils.ResponseWasNotFound(existing.Response()) {
		return tf.RequiresImportError("azurerm_media_streaming_endpoint", resourceId)
	}
	
	autoStart := utils.Bool(false)
```